src/src.js

```diff
@@ -43,19 +45,154 @@ class Game {
+            map[i].push(j);
+        }
+        return map;
+    }
+
+
+    populateMap() {
+
+        // 1. generate an empty map
+
+        let col = 10;  // map size
+        let row = 10;
+        let array2d = this.generateMap(col, row);
+
+        // 2. seed map with initial elements
+
+        let landscapeElements = ['t', 'M', '.']; // trees, mountains, plains (?)
+        let numberOfUniqueElements = ((col * row) / (col + row));  // sparse initial seeding
+        let xSeeded = [];  // track seed locations
+        let ySeeded = [];
+
+        for (let i = 0; i < numberOfUniqueElements; i++) {
+
+            // find random location on map
+            let xSeed = Math.floor(Math.random() * col);
+            let ySeed = Math.floor(Math.random() * row);
+
+            // get random element from available elements
+            let randomElementIndex = Math.floor(Math.random() * landscapeElements.length);
+            let randomElement = landscapeElements[randomElementIndex];
+            
+            // inject element into map
+
+            if (array2d[xSeed][ySeed] !== NaN) {  // if location !alreadySeeded
+                array2d[xSeed][ySeed] = randomElement;
+                xSeeded.push(xSeed);  // track seed locations
+                ySeeded.push(ySeed);
+            }
+
+        }
+
+
+        // 2. grow element territories from seeds
+
+
+        let mapPopulated = false;   // loop until entire map is filled
+        // while (!mapPopulated) {
+        // }
+
+
+
+        function growLoop(xIncrement, yIncrement) { // add probability arg (?)
+
+
+            for (let j = 0; j < xSeeded.length; j++) {
+                let probability = Math.floor(Math.random() * 3);  // likelihood of growing from seed
+                let x = xSeeded[j];
+                let y = ySeeded[j];
+                let growSource = array2d[x][y]; // find a seeded location
+
+                if (array2d[x + xIncrement] === undefined) {  // returns undefined if try seeding beyond map edges
+                    break  // would rather not break ... just skip to the next one
+    
+                } else if (array2d[x + xIncrement][y + yIncrement] !== NaN) {
+                    if (probability === 0) { // 66% chance
+                        console.log('probability check ' + j);
+                        array2d[x + xIncrement][y + yIncrement] = growSource;  // assign neighboring location seed
+                        xSeeded.push(x + xIncrement); // adding the new seed location
+                        ySeeded.push(y + yIncrement);
+                    }
+                }
+            }
+        }
+
+
+        // seed a cell (x,y) from initial seed
+        // idea is to move to neighboring cells to have contained shapes
+
+        growLoop(-1,1);
+        growLoop(-1,-1);
+        growLoop(1,-1);
+        growLoop(1,1);
+        // growLoop(0,1); // motion in just one direction freezes browser?
+        // growLoop(1,0);
```

Comment:
this is my main concern: growLoop(x,y) is supposed to seed a cell x and y steps away from the original cell. this works when either argument is +/-1 (ie. a step diagonally from the initial seed). however, when I pass growLoop() zero as one of the arguments (a step either laterally or vertically), the browser freezes. can you figure out why that might be the case?

Reply:
aahhh ok cool! I'll have to run the code. maybe I'll use this opportunity to add in a module system and a testing framework :)

Reply:
I should have time tomorrow morning